What's new in Swift April 2026 blog post (#1401)
apple/swift-org-website: _posts/2026-05-01-whats-new-in-swift-march-2026.md

---
layout: new-layouts/post
published: true
date: 2026-05-01 10:30:00
title: "What's new in Swift: April 2026 Edition"
author: [adam-fowler, davelester]
category: "Digest"
---

Welcome to "What's new in Swift," a curated digest of releases, videos, and discussions in the Swift project and community.

The 1.0 release of valkey-swift was recently announced on the Valkey blog. We've invited one of the authors to be this month's guest contributor:

> > Hi, I'm Adam Fowler an open source developer working in the swift on server ecosystem. I am excited to announce the 1.0 release of [valkey-swift](https://github.com/valkey-io/valkey-swift) - a production-grade Swift client for [Valkey](https://valkey.io).

Author: ```suggestion
> Hi, I'm Adam Fowler, an open source developer working in the swift on server ecosystem. I am excited to announce the 1.0 release of [valkey-swift](https://github.com/valkey-io/valkey-swift) - a production-grade Swift client for [Valkey](https://valkey.io).
```

>

> > Valkey is a high performance datastore commonly used as a caching layer or message broker in server applications. It is an open source fork of Redis. 

Author: ```suggestion
> Valkey is a high-performance datastore commonly used as a caching layer or message broker in server applications. It is an open source fork of Redis. 
```

> 

> > Valkey-swift is a client library targeted at Valkey servers but is equally capable of working with Redis. It is built from the ground up with Swift 6 and structured concurrency. Every Valkey command returns typed responses checked at compile time, and strict concurrency checking is enabled throughout so that data races are caught by the compiler, not in production. Connection and subscriptions are all scoped through structured concurrency, so resources clean up automatically.

Reviewer: ```suggestion
> Valkey-swift is a client library targeted at Valkey servers but is equally capable of working with Redis. It is built from the ground up with Swift 6 and structured concurrency. Every Valkey command returns typed responses checked at compile time, and strict concurrency checking is enabled throughout so that data races are caught by the compiler, not in production. Connections and subscriptions are all scoped through structured concurrency, so resources clean up automatically.
```

> 
> The client covers every standard Valkey command, auto-generated from Valkey's own command specifications to stay in sync as the server evolves. 
> 

> > Previously the de-facto client library for Redis was RediStack. But it was built on top of pre-concurrency concepts, retro-fitting structured concurrency would have been awkward and some of the new features in valkey-swift infeasible. Also at the same time Redis changed its licensing structure and the open source fork Valkey was created. So it felt a good time to make a break and build a new library.

Author: ```suggestion
> Previously, the de facto client library for Redis was RediStack, which was built on top of pre-concurrency concepts. Retrofitting structured concurrency would have been awkward and some of the new features in valkey-swift infeasible. Around the same time Redis changed its licensing structure and the open source fork Valkey was created. So it felt like a good time to make a clean break and build a new library.
```

Reviewer: should "de-facto" be hyphenated? I think it's just "de facto" - no hyphenation.

> 

> > If you're building server-side Swift and need a fast key-value store, add valkey-swift via Swift Package Manager and you're ready to go. If you are using RediStack to connect with a Redis server we have a [guide](https://swiftpackageindex.com/valkey-io/valkey-swift/1.3.0/documentation/valkey/migrating-from-redistack) to help you migrate to valkey-swift. Complete [documentation is available](https://swiftpackageindex.com/valkey-io/valkey-swift/documentation/valkey), and contributions are welcome on [GitHub](https://github.com/valkey-io/valkey-swift).

Reviewer: ```suggestion
> If you're building server-side Swift and need a fast key-value store, add valkey-swift via Swift Package Manager, and you're ready to go. If you are using RediStack to connect with a Redis server, we have a [guide](https://swiftpackageindex.com/valkey-io/valkey-swift/1.3.0/documentation/valkey/migrating-from-redistack) to help you migrate to valkey-swift. Complete [documentation is available](https://swiftpackageindex.com/valkey-io/valkey-swift/documentation/valkey), and contributions are welcome on [GitHub](https://github.com/valkey-io/valkey-swift).
```

Now on to other news about Swift:

## Videos to watch
* The try! Swift Tokyo 2026 conference featured not one, but **two** exciting talks about Embedded Swift:
    * [Getting started with Embedded Swift](https://www.youtube.com/watch?v=0OcOFBe0jbQ) is a short and accessible introductory talk. Learn about writing Swift using embedded simulators, along with code examples to run Swift code on devices including the Game Boy Advance!
    * [Learn by Building: Bare-Metal Programming with Embedded Swift](https://www.youtube.com/watch?v=vDQ_yxD6JZ4) is a deeper look at Embedded Swift. Follow along by trying the [sample code for five bare-metal Raspberry Pi Pico examples](https://github.com/kishikawakatsumi/tryswift2026) that are featured in the talk.
* Want to learn more about Swift concurrency from engineers who have designed and used its features? Check out this [live online Q&A on Swift concurrency](https://www.youtube.com/watch?v=E95agtPgaa0).
* Nil Coalescing published a new video on some of the lesser known options for working with optionals in [Advanced Techniques for Working with Optionals in Swift](https://www.youtube.com/watch?v=qgDIOrKnmuw).

## New package releases

> * [IndustrialKit](https://github.com/MalkarovPark/IndustrialKit) is a framework for building to design, program, and control robotic systems that was recently [discussed on the Swift forums](https://forums.swift.org/t/industrialkit-a-swift-framework-for-design-and-control-of-robotic-means-of-production/86168).

Reviewer: ```suggestion
* [IndustrialKit](https://github.com/MalkarovPark/IndustrialKit) is a framework for designing, programing, and controlling robotic systems that was recently [discussed on the Swift forums](https://forums.swift.org/t/industrialkit-a-swift-framework-for-design-and-control-of-robotic-means-of-production/86168).
```

"building to design.." seems kind of awkward, what do you think of reframing this using gerunds?

* [swift-tar](https://forums.swift.org/t/swift-tar-a-pure-swift-tar-archive-read-write-extract-library/85844) is a pure Swift library for reading, writing, and extracting TAR archives. swift-tar is cross-platform, works without requiring any system framework or Foundation, and supports GNU & PAX extensions.

> * [Xylem](https://github.com/compnerd/xylem) is pure Swift XML parser with zero dependencies, covering SAX, DOM, and XPath 1.0.

Reviewer: ```suggestion
* [Xylem](https://github.com/compnerd/xylem) is a pure Swift XML parser with zero dependencies, covering SAX, DOM, and XPath 1.0.
```
